[IV-22-23] Objetivo 7

iscoct opened the pull request:
* [x] ¿Se han registrado todos los eventos relevantes en la aplicación, es decir, se han añadido comandos para registrarlos en todos ellos?
* [x]  ¿Se han seguido buenas prácticas en el uso de configuración remota?
* [x]  ¿Se han explicado correctamente las decisiones técnicas que se han tomado sobre las bibliotecas usadas?
* [x]  ¿Te has asegurado de que el fichero `.env` no se pueda añadir accidentalmente al repositorio?
* [x]  ¿Los logs y la configuración están en un objeto que se _inyecta_ en las clases manejadoras para su uso?
* [x]  ¿_Realmente_ se usan los logs en las diferentes clases en eventos importantes?
* [x]  ¿Se testea el código creado para logs y configuración, igual que cualquier otro código?
* [x]  ¿La clase o módulo de configuración lee _siempre_ las variables de entorno?

Commit:
Seleccionado e investigado las librerias

Se escoge la librerias logs
Si se quiere mas informacion se puede consultar la documentacion
asociada a este commit
Closes #52

Relacionado con #53 por documentar el proceso de inyeccion de deps.

Commit:
Se imprime los logs cuando se ejecutan los tests

Relacionado con #53

Commit:
Implementado wrapper para la libreria logs

Usada tecnicas de inyeccion de dependencias segun las
buenas practicas de Go

Closes #53 #52

Commit:
Añadido configuracion de app Close #53

Commit:
Adding ROOT_PATH a github actions

Commit:
adding ROOT_PATH a Azure Pipelines

Commit:
Printing current dir in Azure Pipelines

Commit:
Cambiando el valor de ROOT_PATH

Commit:
Testing only my repo in Azure Pipelines

Commit:
Trying again to execute Azure Pipeliens correctly

Commit:
Executing again azure

Commit:
Running Azure pipeline again

Commit:
Running azure pipelines again

Commit:
Adding verbosity to test in azure devops

Commit:
Adding ROOT_PATH to semaphore pipeline

Commit:
Adding on push to github

Commit:
Running pipeline again

Commit:
Running pipeline again

Commit:
Running pipeline again

Commit:
Running pipeline again

Commit:
Running pipeline again

Commit:
Running pipeline again

Commit:
Running pipeline again

Commit:
Adding ROOT_PATH to Taskfile

Commit:
Added configuracion field to iv.yaml

Closes #53

Commit:
Added config folder to .gitignore

Closes #53

Commit:
Removed basic_conf.json from repo

Added more rules in gitignore

Closes #53

Commit:
Wrote explicitly config file closes #53

Commit:
Added default parameters when there is no config

Closes #53

Commit:
Changed viper por godotenv

Se lee ahora la configuración a través de .env en
vez de un json

Closes #53

Commit:
Removed ROOT_PATH from pipelines

Closes #53

JJ commented on iv.yaml:
```diff
@@ -12,6 +12,8 @@ test:
   - types/models/comanda_test.go
   - types/models/optimizador_test.go
 
+configuracion: .env
```
Vamos a ver, si se trata de que no lo añadáis en el repo no es para que lo pongas aquí... 
Lo que tienes que poner es el fichero donde has programado la configuración.

iscoct replied:
Cambiado y creada la abstracción de la lectura de la configuración

JJ reviewed (changes requested):
Ni has programado la configuración, como te he dicho en el grupo de Telegram, ni has justificado la elección de una biblioteca que la lea u otra, ni has puesto correctamente la configuración del iv.yaml. Por favor, léete atentamente el guión y trata de hacerlo correctamente en esta ocasión.

Commit:
Documentado criterio y selected libreria config

Documentado el criterio escogido para la seleccion
de la libreria de configuracion y la seleccion final que
se ha hecho

Closes #55

Commit:
Abstraida lectura de conf con lib seleccionada

Closes #56

iscoct commented:
Buenas @JJ ya hice todos los cambios que me pediste.

Se ha creado una abstracción para la lectura de la configuración, documentada distintas librerías de configuración y el valor de configuracion en iv.yaml

Estoy a la espera para solucionar posibles otros problemas.

JJ commented on Taskfile.yml:
```diff
@@ -7,17 +7,17 @@ tasks:
     cmds:
-      - cp -R ./types ../ejecucion_tests
-      - cd ../ejecucion_tests && go mod tidy && go test ./...
+      - cp -R . ../ejecucion_tests
```
??

iscoct replied:
Eso es porque antes sólo necesitaba que se copiara la carpeta ./types que es donde tenía el código.

Como ahora necesito también la carpeta ./cmd/* he copiado la carpeta entera para preservar mi estructura.

iscoct replied:
Puedo hacer

- cp -R ./types ../ejecucion_tests/types
- cp -R ./cmd ../ejecucion_tests/cmd

Ya me dice si le parece mejor.

JJ replied:
Se trata de que el fuente se *monte* en el contenedor, no que copies cosas. Si estás haciendo esto, estás incurriendo en alguna mala práctica en algún lugar.

iscoct replied:
Llevas razón, pero como vimos en el objetivo 5, necesitaba copiarlo porque no podía modificar nada del código porque estaba en modo lectura y al instalar las dependencias, se modificaba el go.sum porque tiene ahí un hash con información de la instalación de los paquetes (al instalarlo otra vez y no poder modificar el go.sum daba error).

En resumen, que tenía que copiarlo en una carpeta distinta nueva, en este caso decidí que fuera ejecucion_tests.

¿Le parece bien entonces que lo haga de esa manera o lo dejo?
Si tiene alguna otra idea la comentamos @JJ

iscoct replied:
Llevas razón, pero como vimos en el objetivo 5, necesitaba copiarlo porque no podía modificar nada del código porque estaba en modo lectura y al instalar las dependencias, se modificaba el go.sum porque tiene ahí un hash con información de la instalación de los paquetes (al instalarlo otra vez y no poder modificar el go.sum daba error).

En resumen, que tenía que copiarlo en una carpeta distinta nueva, en este caso decidí que fuera ejecucion_tests.

Lo dejo así para cerrar el objetivo entonces.

Si tiene alguna otra idea la comentamos @JJ

JJ commented on cmd/pkg/config/conf_reader.go:
```diff
@@ -0,0 +1,37 @@
+var instance *confReader
+
+func createConfReader() {
+	ip := "127.0.0.1"
```
Siempre debes usar constantes en vez de literales. En este caso, "DEFAULT_" habría sido lo adecuado.

iscoct replied:
No le he puesto eso porque después la sobrescribo si puede leer la configuración.

Puedo declararlas como constantes y default, luego si puede leer la configuración, setear las variables port e ip y sino puede asignarles las constantes.

Si cree que es mejor me dice.

JJ replied:
No estoy hablando de `ip`, estoy hablando del literal que le asignas. En todo caso, `ip` tampoco es un nombre suficientemente descriptivo.
El objetivo, por otro lado, está ya aprobado. Simplemente te trato de aportar buenas prácticas para tu trabajo diario o para que las hagas antes de cerrar. Es decisión tuya.

JJ commented on cmd/pkg/logger/global.go:
```diff
@@ -0,0 +1,25 @@
+	"cotan/cmd/pkg/config"
+)
+
+var Standard Logger
```
La denominación de las variables es importante. "Standard" no da idea de qué es lo que contiene.

iscoct replied:
¿StandardOutputLogger le parece mejor?

Espero respuesta para cambiarlo.

JJ replied:
Como te he dicho antes, usar nombres descriptivos es una buena práctica, pero no el objetivo en este caso. Haz lo que consideres conveniente.

iscoct replied:
Perfecto, ¡gracias!

JJ reviewed (approved):
:+1:

iscoct commented:
Antes de cerrarlo espero la respuesta a mis sugerencias de los cambios que puedo hacer y saber cuales ve necesarios @JJ
